components/nexusoperations/executors.go

```diff
@@ -205,10 +204,35 @@ func (e taskExecutor) executeInvocationTask(ctx context.Context, env hsm.Environ
 		return fmt.Errorf("%w: %w", queues.NewUnprocessableTaskError("failed to generate a callback token"), err)
 	}
 
-	callCtx, cancel := context.WithTimeout(
-		ctx,
-		e.Config.RequestTimeout(ns.Name().String(), task.EndpointName),
-	)
+	// The following logic handles the operation ScheduleToCloseTimout parameter and Operation-Timeout header.
+	// The minimum of the resolved operation timeout and the configured request timeout is used for the context timeout
+	// when making the StartOperation request, according to the following logic:
+	// (ScheduleToClose set, Operation-Timeout set) -> no changes, use ScheduleToClose
```

Comment:
```suggestion
	// (ScheduleToClose set, Operation-Timeout set) -> Use Operation-Timeout
```

Reply:
I thought we discussed that `ScheduleToClose` should always be used if it is set.

Reply:
I mean just the `Operation-Timeout` header, not the context deadline.